File: Creational/Builder/GoLang/burger_meal_builder.go
package main

type BurgerMealBuilder struct {
	meal *Meal
}

func NewBurgerMealBuilder() *BurgerMealBuilder {
	return &BurgerMealBuilder{meal: &Meal{}}
}

func (b *BurgerMealBuilder) BuildMainItem() {
	b.meal.MainItem = "Burger"
}

func (b *BurgerMealBuilder) BuildSideItem() {
	b.meal.SideItem = "Fries"
}

func (b *BurgerMealBuilder) BuildDrink() {
	b.meal.Drink = "Coke"
}

func (b *BurgerMealBuilder) BuildToy() {
	b.meal.Toy = "Toy Car"
}

func (b *BurgerMealBuilder) GetMeal() *Meal {
	return b.meal
}

File: Creational/Builder/C#/BurgerMealBuilder.cs
namespace FastFoodBuilder {
    public class BurgerMealBuilder : IMealBuilder {
        private Meal _meal = new Meal();

        public void BuildMainItem() {
            _meal.MainItem = "Burger";
        }

        public void BuildSideItem() {
            _meal.SideItem = "Fries";
        }

        public void BuildDrink() {
            _meal.Drink = "Coke";
        }

        public void BuildToy() {
            _meal.Toy = "Toy Car";
        }

        public Meal GetMeal() {
            return _meal;
        }
    }
}
